cmd/uplink/cmd/root.go

```diff
@@ -38,8 +39,46 @@ var GWCmd = &cobra.Command{
 	Short: "The Storj client-side S3 gateway",
 }
 
+func applicationDir(subdir ...string) string {
+	for i := range subdir {
+		if runtime.GOOS == "windows" || runtime.GOOS == "darwin" {
+			subdir[i] = strings.Title(subdir[i])
+		} else {
+			subdir[i] = strings.ToLower(subdir[i])
+		}
+	}
+	home := os.Getenv("HOME")
+	var appdir string
+
+	switch runtime.GOOS {
+	case "windows":
+		// Windows standards: https://msdn.microsoft.com/en-us/library/windows/apps/hh465094.aspx?f=255&MSPPError=-2147217396
+		for _, env := range []string{"AppData", "AppDataLocal", "UserProfile", "Home"} {
+			val := os.Getenv(env)
+			if val != "" {
+				appdir = val
+				break
+			}
+		}
+	case "darwin":
+		// Mac standards: https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html
+		appdir = filepath.Join(home, "Library", "Application Support")
+	case "linux":
+		fallthrough
+	default:
+		// XDG standards: https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html
```

Comment:
I think you should use the same structure here that you use in the "windows" case:
`for _, env := range []string{"XDG_DATA_HOME", "HOME"} {`
And get rid of the `home := os.Getenv("HOME")` outside of the switch statement. Move that line to the "darwin" case.